bin/cron/build_contact_rollups

```diff
@@ -0,0 +1,432 @@
+  PEGASUS_DB.run "
+    INSERT INTO contact_rollups (email, name, dashboard_user_id, roles, city, state, postal_code, country)
+    SELECT email COLLATE utf8_general_ci, name, users.id, 'Teacher', city, state, postal_code, country FROM #{DASHBOARD_DB_NAME}.users AS users
+    LEFT OUTER JOIN #{DASHBOARD_DB_NAME}.user_geos AS user_geos ON user_geos.user_id = users.id
+    WHERE users.user_type = 'teacher' AND LENGTH(email) > 0 AND user_geos.indexed_at IS NOT NULL
+    ON DUPLICATE KEY UPDATE contact_rollups.name = VALUES(name), contact_rollups.dashboard_user_id = VALUES(dashboard_user_id), roles = VALUES(roles)"
+  log_completion(start)
+end
+
+def insert_from_dashboard_pd_enrollments
+  start = Time.now
+  log "Inserting contacts from dashboard.pd_enrollments"
+  PEGASUS_DB.run "
+    INSERT INTO contact_rollups (email, name)
+    SELECT email, name FROM #{DASHBOARD_DB_NAME}.pd_enrollments AS pd_enrollments
+    WHERE LENGTH(pd_enrollments.email) > 0
+    ON DUPLICATE KEY UPDATE name = contact_rollups.name"
+  log_completion(start)
+end
+
+def insert_from_pegasus_contacts
+  start = Time.now
+  log "Inserting contacts from pegasus.contacts"
+  PEGASUS_DB.run "
+    INSERT INTO contact_rollups (email, opted_out, name)
+    SELECT email, IF(unsubscribed_at IS null, null, true) AS opted_out, name FROM contacts WHERE LENGTH(email) > 0
+    ON DUPLICATE KEY UPDATE contact_rollups.name = VALUES(name)"
+  log_completion(start)
+end
+
+def insert_from_pegasus_forms
+  start = Time.now
+  log "Inserting contacts and IP geo data from pegasus.forms"
+  PEGASUS_DB.run "
+    INSERT IGNORE INTO contact_rollups (email, name, city, state, postal_code, country)
+    SELECT email, name, city, state, postal_code, country FROM forms
+    LEFT OUTER JOIN form_geos on form_geos.form_id = forms.id
+    WHERE form_geos.indexed_at IS NOT NULL"
+  log_completion(start)
+end
+
+def update_roles
+  start = Time.now
+  log "Updating Facilitator, Workshop Organizer, District Contact roles from dashboard.users"
+  append_to_role_list_from_permission("facilitator", "Facilitator")
+  append_to_role_list_from_permission("workshop_organizer", "Workshop Organizer")
+  append_to_role_list_from_permission("district_contact", "District Contact")
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+
+  start = Time.now
+  log "Updating Professional Learning Partner role"
+  append_plp_to_role_list
+  log_completion(start)
+end
+
+def append_to_role_list_from_permission(permission_name, dest_value)
+  PEGASUS_DB.run "
+    UPDATE contact_rollups
+    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contact_rollups.dashboard_user_id
+    INNER JOIN #{DASHBOARD_DB_NAME}.user_permissions AS user_permissions ON user_permissions.user_id = users.id
+    SET roles = CONCAT(COALESCE(CONCAT(roles, ','), ''), '#{dest_value}')
+    WHERE LENGTH(users.email) > 0
+    AND user_permissions.permission = '#{permission_name}' AND contact_rollups.id > 0"
```

Comment:
This was originally to make the update work even in SQL_SAFE_UPDATE mode. I later caved and set SQL_SAFE_UPDATE off for the connection. I can remove these. I will do that in a future PR as it requires a re-test cycle.